kiwix-desktop.pro

```diff
@@ -20,12 +20,22 @@ CONFIG += link_pkgconfig
 
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 
-TARGET = kiwix-desktop
+TARGET = kiwix
 TEMPLATE = app
 
 QMAKE_CXXFLAGS += -std=c++17
 QMAKE_LFLAGS +=  -std=c++17
 
+# Kiwix requires per default on Qt6 (but can still compile with Qt5 if explicitly
+# requested, see README.md for more information).
+QT_SELECT += $$(QT_SELECT)
+lessThan(QT_MAJOR_VERSION, 6) {
+    !equals(QT_SELECT, "qt5") {
```

Comment:
Do you want a kiwix-specific env var to control the build behavior? The `QT_SELECT` env var is not the only way to select which Qt installation to use, so someone might be intentionally using a qt5 toolchain yet the build would fail because the `QT_SELECT` env var was not set. Similarly, if someone already set `QT_SELECT=qt5` as part of their default shell environment, the build _wouldn't_ fail even though maybe it should have.

A kiwix-specific var like `KIWIX_QT=qt5` would prevent this.

Reply:
I have no strong opinion on this point... I mostly focus on simplicity. Just to be sure I understand your scenarios:

> someone might be intentionally using a qt5 toolchain yet the build would fail because the QT_SELECT env var was not set.

OK, but then he clearly ignores the `README` right? If he ignores the `README` he won't find out how to set `KIWIX_QT=qt5` either.

> Similarly, if someone already set QT_SELECT=qt5 as part of their default shell environment, the build wouldn't fail even though maybe it should have.

You mean, he would have set `QT_SELECT=qt5` but actually compiling with Qt6 for example? OK... but that sounds pretty weird!